cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/runtime/service/http/BasicHttpServiceContext.java

```diff
@@ -253,4 +259,25 @@ public void releaseCallResources() {
 
     return evaluated;
   }
+
+  /**
+   * Get preferences for the supplied namespace.
+   *
+   * If the namespace doesn't exist, we return null.
+   *
+   * @param namespace the name of the namespace to fetch preferences for.
+   * @param resolved true if resolved properties are desired.
+   * @return Map containing the preferences for this namespace
+   * @throws IOException if the preferencesFetcher could not complete the request.
+   */
+  @Override
+  public Map<String, String> getPreferencesForNamespace(String namespace, boolean resolved) throws IOException {
+    try {
+      return preferencesFetcher.get(new NamespaceId(namespace), resolved).getProperties();
+    } catch (NotFoundException nfe) {
+      LOG.warn(String.format("Namespace %s not found ", namespace));
```

Comment:
shouldn't log a warning here, as it might not be unexpected, depending on how the caller is using it.

Reply:
I'll remove this log.